GeeksCoreLibrary/Components/Configurator/Configurator.cs

```diff
@@ -519,16 +458,141 @@ public override async Task<HtmlString> InvokeAsync(DynamicContent dynamicContent
+
+            // Build the main HTML.
+            var mainHtml = await TemplatesService.DoReplacesAsync(configuratorData.MainTemplate, removeUnknownVariables: false);
+            mainHtml = mainHtml.Replace("{mainsteps}", stepsHtml, StringComparison.OrdinalIgnoreCase)
+                .Replace("{steps}", stepsHtml, StringComparison.OrdinalIgnoreCase)
+                .Replace("{substeps}", stepsHtml, StringComparison.OrdinalIgnoreCase);
+
+            mainHtml = mainHtml.Replace("{progressbar}", progressBarHtml, StringComparison.OrdinalIgnoreCase)
+                .Replace("{progress}", progressBarHtml, StringComparison.OrdinalIgnoreCase);
+
+            mainHtml = mainHtml.Replace("{summary}", summaryHtml, StringComparison.OrdinalIgnoreCase);
+
+            var resultHtml = new StringBuilder();
+            resultHtml.Append("<div id=\"configurator\" v-cloak>");
+            resultHtml.Append(mainHtml);
+            resultHtml.Append("</div>");
+
+            return resultHtml.ToString();
+        }
+
+        /// <summary>
+        /// Renders the HTML of a step in Vue mode.
+        /// </summary>
+        /// <param name="configuratorData">The full configurator data.</param>
+        /// <param name="stepId">The ID of the step to render.</param>
+        /// <returns>The rendered HTML of a step in Vue mode.</returns>
+        /// <exception cref="Exception">If a step with the given <paramref name="stepId"/> is not found.</exception>
+        private async Task<string> RenderVueStepAsync(VueConfiguratorDataModel configuratorData, ulong stepId)
+        {
+            var stepData = configuratorData.StepsData.SingleOrDefault(s => s.StepId == stepId);
+            if (stepData == null)
+            {
+                throw new Exception($"Step with ID {stepId} not found.");
+            }
+
+            var template = stepData.StepTemplate;
+            template = await TemplatesService.DoReplacesAsync(template, removeUnknownVariables: false);
+
+            // Render sub steps.
+            if (template.Contains("{options}") || template.Contains("{datasource_values}"))
+            {
+                // Render step option template HTML.
+                var stepOptionHtmlBuilder = new StringBuilder();
+                stepOptionHtmlBuilder.Append("<step-option v-for=\"option in step.options\" :key=\"option.id\">");
+                stepOptionHtmlBuilder.Append(await TemplatesService.DoReplacesAsync(stepData.StepOptionTemplate, removeUnknownVariables: false));
+                stepOptionHtmlBuilder.Append("</step-option>");
+                var stepOptionHtml = stepOptionHtmlBuilder.ToString();
+
+                template = template.Replace("{options}", stepOptionHtml)
+                    .Replace("{datasource_values}", stepOptionHtml, StringComparison.OrdinalIgnoreCase);
+            }
+            if (template.Contains("{steps}") || template.Contains("{substeps}"))
```

Comment:
Could you add an empty line above this if statement, so that it's clearer on first glance that this is a separate statement and not an elseif?